File: DevExpress-Demo/CS/NavigationDemo.Wpf/ViewModels/NavigationPane/MailViewModel.cs
using System.Collections.Generic;
using System.Collections.ObjectModel;
using DevExpress.Mvvm;

namespace NavigationDemo {
    public class MailViewModel : ViewModelBase {
        public virtual IList<MailItem> ItemsSource { get; set; }

        public virtual int SentGroupIndex { get; set; }
        public virtual int ReceiveGroupIndex { get; set; }

        public virtual NavigationId FolderId { get; set; }

        public MailViewModel() {
            SentGroupIndex = -1;
            ReceiveGroupIndex = -1;
        }
        protected override void OnParameterChanged(object parameter) {
            if(parameter != null)
                UpdateIds((NavigationId)parameter);
        }
        void UpdateIds(NavigationId id) {
            FolderId = id;
            switch(id) {
                case NavigationId.Inbox: SentGroupIndex = -1; ReceiveGroupIndex = 0; break;
                case NavigationId.Outbox: SentGroupIndex = 0; ReceiveGroupIndex = -1; break;
                case NavigationId.SentItems: SentGroupIndex = 0; ReceiveGroupIndex = -1; break;
                case NavigationId.DeletedItems: SentGroupIndex = -1; ReceiveGroupIndex = 0; break;
                case NavigationId.Drafts: SentGroupIndex = 0; ReceiveGroupIndex = -1; break;
            }
            ItemsSource = NavigationPaneData.MailData[id];
        }
    }
}

File: DevExpress-Demo/VB/NavigationDemo.Wpf/ViewModels/NavigationPane/MailViewModel.vb
Imports System.Collections.Generic
Imports DevExpress.Mvvm

Namespace NavigationDemo

    Public Class MailViewModel
        Inherits ViewModelBase

        Public Overridable Property ItemsSource As IList(Of MailItem)

        Public Overridable Property SentGroupIndex As Integer

        Public Overridable Property ReceiveGroupIndex As Integer

        Public Overridable Property FolderId As NavigationId

        Public Sub New()
            SentGroupIndex = -1
            ReceiveGroupIndex = -1
        End Sub

        Protected Overrides Sub OnParameterChanged(ByVal parameter As Object)
            If parameter IsNot Nothing Then UpdateIds(CType(parameter, NavigationId))
        End Sub

        Private Sub UpdateIds(ByVal id As NavigationId)
            FolderId = id
            Select Case id
                Case NavigationId.Inbox
                    SentGroupIndex = -1
                    ReceiveGroupIndex = 0
                Case NavigationId.Outbox
                    SentGroupIndex = 0
                    ReceiveGroupIndex = -1
                Case NavigationId.SentItems
                    SentGroupIndex = 0
                    ReceiveGroupIndex = -1
                Case NavigationId.DeletedItems
                    SentGroupIndex = -1
                    ReceiveGroupIndex = 0
                Case NavigationId.Drafts
                    SentGroupIndex = 0
                    ReceiveGroupIndex = -1
            End Select

            ItemsSource = MailData(id)
        End Sub
    End Class
End Namespace
